src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -229,7 +230,15 @@ class ErgoMemPool private[mempool](private[mempool] val pool: OrderedTxPool,
               acceptIfNoDoubleSpend(unconfirmedTx, validationStartTime)
           }
         } else {
-          val exc = new Exception(s"Pool can not accept transaction ${tx.id}, it is invalidated earlier or the pool is full")
+          val contains = this.contains(tx.id)
+          val msg = if(contains) {
+            s"Pool can not accept transaction ${tx.id}, it is already in the mempool"
+          } else if(pool.size == settings.nodeSettings.mempoolCapacity) {
+            s"Pool can not accept transaction ${tx.id}, the mempool is full"
+          } else {
+            s"Pool can not accept transaction ${tx.id}"
```

Comment:
we could check `isInvalidated` here to see what's going on

Reply:
added logging in https://github.com/ergoplatform/ergo/pull/1862/commits/381624d12650489f30e5d549a09cf739e6c80c2f